google/auth/identity_pool.py

```diff
@@ -145,9 +148,69 @@ def get_subject_token(self, context, request):
 class _X509Supplier(SubjectTokenSupplier):
     """Internal supplier for X509 workload credentials. This class is used internally and always returns an empty string as the subject token."""
 
+    def __init__(self, trust_chain_path, leaf_cert_callback):
+        self._trust_chain_path = trust_chain_path
+        self._leaf_cert_callback = leaf_cert_callback
+
     @_helpers.copy_docstring(SubjectTokenSupplier)
     def get_subject_token(self, context, request):
-        return ""
+        leaf_cert = crypto.load_certificate(
+            crypto.FILETYPE_PEM, self._leaf_cert_callback()
+        )
+        trust_chain = self._read_trust_chain()
+        cert_chain = []
+
+        cert_chain.append(_X509Supplier._encode_cert(leaf_cert))
+        for i, cert in enumerate(trust_chain):
+            encoded = _X509Supplier._encode_cert(cert)
+            # Check if the current cert is the leaf cert and raise an exception if it is not the first
+            # cert in the chain.
+            if encoded == cert_chain[0]:
```

Comment:
it could be more efficient and easy to understand if this check is outside the loop? You can just chck whether first element of trust chain and leaf are same?